Core/PARStore.m

```diff
@@ -949,6 +950,85 @@ - (void)setEntriesFromDictionary:(NSDictionary *)dictionary
      }];
 }
 
+- (void)appendChanges:(NSArray *)changes forDeviceIdentifier:(NSString *)deviceIdentifier
+{
+    // Model and PSC
+    NSManagedObjectModel *mom = [PARStore managedObjectModel];
+    NSPersistentStoreCoordinator *psc = [[NSPersistentStoreCoordinator alloc] initWithManagedObjectModel:mom];
+    
+    // Persistent Store
+    NSError *error = nil;
+    NSString *dirPath = [self directoryPathForDeviceIdentifier:deviceIdentifier];
+    [[NSFileManager defaultManager] createDirectoryAtPath:dirPath withIntermediateDirectories:NO attributes:nil error:NULL];
+    [self addPersistentStoreWithCoordinator:psc dirPath:dirPath readOnly:NO error:&error];
+    
+    // Context
+    NSManagedObjectContext *moc = [[NSManagedObjectContext alloc] initWithConcurrencyType:NSPrivateQueueConcurrencyType];
+    [moc setPersistentStoreCoordinator:psc];
+    [moc setUndoManager:nil];
+
+    // Have main context observe changes made in background, and merge them.
+    id observer = [[NSNotificationCenter defaultCenter] addObserverForName:NSManagedObjectContextDidSaveNotification object:moc queue:nil usingBlock:^(NSNotification *notification) {
+        [self.databaseQueue dispatchSynchronously:^{
+            [[self managedObjectContext] mergeChangesFromContextDidSaveNotification:notification];
+        }];
+    }];
+    
+    // Add changes
+    [moc performBlockAndWait:^{
+        NSError *error = nil;
+        
+        // Expression to retrieve the maximum timestamp in the store
```

Comment:
I thought we had discussed that we cannot rely on this in order to make a robust system where changes might be added not in order?

My initial thought when seeing the expression based stuff is, why --if we any way need to rely on order of insertion-- are we not simply fetching the last row in the db?

However, I thought we discussed how we will need to be robust against:
a) receiving batches of changes out of order, and
b) be resilient against batches containing the overlapping changes

Both of these you would (have to) address by checking for each change if the row is in the db already or not, more expensive but most reliable.

Question is then, would we need to include also the blob in that check (is timestamp + key enough, or should the query be timestamp, key, and blob being the same), @cparnot? If we need blob to also be part we might have to add an index I guess.